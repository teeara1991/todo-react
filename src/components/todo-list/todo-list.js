import React from "react";
import {TodoItem} from '../todo-item/todo-item';
export function TodoList(props){
  return (
    <ul className="todos todos-custom">
          {props.todos.map((todo, index) => (
            <TodoItem
              key={index}
              name={todo.name}
              checked={todo.checked}
              completed={todo.completed}
              onDelete={() => props.onDelete(index)}
              onComplete={() => props.onComplete(index)}/>
          ))}
    </ul>
  );
}