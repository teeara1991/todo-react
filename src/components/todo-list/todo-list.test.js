import React from "react";
import { TodoList } from "../todo-list/todo-list";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

it("calls removeTodo Redux action creator with trash click", () => {
  const props = {
    onDelete: jest.fn(),
    todos: [
      { name: "Buy groceries", completed: false },
      { name: "Change oil", completed: false }
    ]
  };
  const wrapper = mount(<TodoList {...props} />);
  wrapper
    .find(".fas")
    .at(0)
    .simulate("click");
  expect(props.onDelete).toHaveBeenCalledWith(0);
});
it("calls completeTodo Redux action creator with input click", () => {
  const props = {
    onComplete: jest.fn(),
    todos: [
      { name: "Buy groceries", completed: false },
      { name: "Change oil", completed: false }
    ]
  };
  const wrapper = mount(<TodoList {...props} />);
  wrapper
    .find(".todo-text")
    .at(0)
    .simulate("click");
  wrapper
    .find("input")
    .at(0)
    .simulate("click");
  expect(props.onComplete).toHaveBeenCalledWith(0);
});

it("matches the snapshot", () => {
  const props = {
    todos: [
      { name: "Buy groceries", completed: false },
      { name: "Change oil", completed: false }
    ]
  };
  const tree = shallow(<TodoList {...props} />);
  expect(toJson(tree)).toMatchSnapshot();
});
