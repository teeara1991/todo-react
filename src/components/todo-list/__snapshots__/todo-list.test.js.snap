// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<ul
  className="todos todos-custom"
>
  <TodoItem
    completed={false}
    key="0"
    name="Buy groceries"
    onComplete={[Function]}
    onDelete={[Function]}
  />
  <TodoItem
    completed={false}
    key="1"
    name="Change oil"
    onComplete={[Function]}
    onDelete={[Function]}
  />
</ul>
`;
